services/src/atdd-staging/src/main/java/org/openkilda/atdd/staging/service/northbound/NorthboundServiceImpl.java

```diff
@@ -184,6 +195,46 @@ public RulesValidationResult validateSwitchRules(String switchId) {
                 .collect(Collectors.toList());
     }
 
+    @Override
+    public List<LinkPropsDto> getLinkProps(LinkPropsDto keys) {
```

Comment:
LinkPropsDto can handle more than src/dest switch/port. Why should we pass the whole object instead of 4 fields?

Reply:
This is implemented similar to the original 'getLinkProps' in the LinkService in Northbound.
```
    /**
     * These results are not related to the ISL links per se .. they are based on any link
     * properties that have been uploaded through setLinkProps.
     *
     * @param keys the primary keys are used if set, and the props are ignored. If nothing is set, all
     *             link properties rows are returned.
     * @return one or more link properties from the static link_props table
     * */
    List<LinkPropsDto> getLinkProps(LinkPropsDto keys);
```

Reply:
fixed to match current implementation in PR 881